presto-main/src/main/java/com/facebook/presto/SessionRepresentation.java

```diff
@@ -69,6 +71,7 @@ public SessionRepresentation(
             @JsonProperty("userAgent") Optional<String> userAgent,
             @JsonProperty("clientInfo") Optional<String> clientInfo,
             @JsonProperty("clientTags") Set<String> clientTags,
+            @JsonProperty("resourceEstimate") Optional<ResourceEstimates> resourceEstimate,
```

Comment:
Is this only `Optional` for backwards compatibility?  If so, how about we change the assignment line below to:
```
this.resourceEstimate = requireNonNull(resourceEstimate, "resourceEstimate is null").orElse(new ResourceEstimate());
```
In generally, I find it weird to have an Optional of a thing where every nested field is Optional.